src/safeposix/syscalls/net_calls.rs

```diff
@@ -721,16 +759,86 @@ impl Cage {
         }
     }
 
-    //The connect() system call connects the socket referred to by the
-    //file descriptor fd to the address specified by remoteaddr.
+    /// ### Description
+    /// 
+    /// `connect_syscall` connects the socket referred to by the
+    /// file descriptor fd to the address specified by remoteaddr. 
+    /// 
+    /// ### Arguments
+    /// 
+    /// it accepts two parameters: 
+    /// * `fd` - an open file descriptor
+    /// * `remoteaddr` - the address to request a connection to
+    /// 
+    /// ### Returns
+    /// 
+    /// for a successful call, zero is returned. On
+    /// error, -errno is returned, and errno is set to indicate the error.
+    /// 
+    /// ### Errors
```

Comment:
I think sorting this a bit into errors that we can get from RustPOSIX and errors we can get from the raw libc call would be helpful